CMakeLists.txt

```diff
@@ -273,28 +299,33 @@ if(Legion_BUILD_APPS OR Legion_BUILD_EXAMPLES OR Legion_BUILD_TUTORIAL OR Legion
     set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} $<$<BOOL:${PROP}>:-I$<JOIN:${PROP}, -I>>")
   endif()
 
-  if(Legion_BUILD_APPS)
+  if(Legion_BUILD_ALL OR Legion_BUILD_APPS)
     add_subdirectory(apps)
   endif()
-  if(Legion_BUILD_EXAMPLES)
+  if(Legion_BUILD_ALL OR Legion_BUILD_EXAMPLES)
     add_subdirectory(examples)
   endif()
-  if(Legion_BUILD_TUTORIAL)
+  if(Legion_BUILD_ALL OR Legion_BUILD_TUTORIAL)
     add_subdirectory(tutorial)
   endif()
-  if(Legion_BUILD_TESTS)
+  if(Legion_BUILD_ALL OR Legion_BUILD_TESTS)
     add_subdirectory(test)
   endif()
 endif()
 
 #------------------------------------------------------------------------------#
-# configure header
+# Configure definition headers
 #------------------------------------------------------------------------------#
 
-configure_file(${PROJECT_SOURCE_DIR}/cmake/legion_defines.h.in ${PROJECT_SOURCE_DIR}/runtime/legion_defines.h @ONLY)
-include_directories(${PROJECT_SOURCE_DIR}/runtime)
-install(FILES ${PROJECT_SOURCE_DIR}/runtime/legion_defines.h DESTINATION include)
+configure_file(${PROJECT_SOURCE_DIR}/cmake/legion_defines.h.in
+  ${PROJECT_SOURCE_DIR}/runtime/legion/legion_defines.h @ONLY)
```

Comment:
The output of `configure_file()` should go into `${PROJECT_BINARY_DIR}` otherwise you could end up in a mess when doing two out of sources builds.